lib/src/digest.dart

```diff
@@ -39,5 +38,6 @@ class Digest {
 
   /// The message digest as a string of hexadecimal digits.
   @override
-  String toString() => hex.encode(bytes);
+  String toString() =>
+      bytes.map((b) => b.toRadixString(16).padLeft(2, '0')).join('');
```

Comment:
That looks fairly inefficient. For a list of length n you create n-2n smaller strings and then join them.

Is `toString` used for anything except debugging? If so, then how about using something like:
```dart
String _toHex(List<int> bytes) {
   const digits = "0123456789abcdef";
   var hex = Uint8List(bytes.length * 2);
   for (var i = 0, j = 0; i < bytes.length; i++) {
     var byte = bytes[i];
     hex[j++] = digits.codeUnitAt((byte & 0xF0) >> 4);
     hex[j++] = digits.codeUnitAt(byte & 0xF);
   }
   return String.fromCharCodes(hex);
}
```